### ISSUE TITLE
Error while checking for vulnerabilities

### ISSUE THREAD

**Issue by todmephis** [drwetter/testssl.sh#1783]
When checking for vulnerabilities I got this error:

` Checking for vulnerable RC4 Ciphers 
 RC4 (CVE-2013-2566, CVE-2015-2808)        /root/Desktop/Tools-Script/testssl.sh/testssl.sh: line 1136: 16#: invalid integer constant (error token is "16#")
/root/Desktop/Tools-Script/testssl.sh/testssl.sh: line 12319: 2+: syntax error: operand expected (error token is "+")
`
It happened with RC4, LUCKY, etc.

I fixed this error by changing line 1136 on file testssl.sh form  `echo $((16#$1))`  to `echo $((0x$1))`

After the change, the problem was fixed.

` Checking for vulnerable RC4 Ciphers 
 RC4 (CVE-2013-2566, CVE-2015-2808)        VULNERABLE (NOT ok): RC4-SHA RC4-MD5 `

**Comment by drwetter** [comment]
Hi @todmephis ,

this is not a general error. And unfortunately there's little information so that I can reproduce that. 

Would you be so kind to provide the details which were asked in the template?

Thx, Dirk

**Comment by eny13** [comment]
I'm experiencing a similar Issue with the exact same error message as @todmephis .
Once testssl reaches check for FREAK, I get the error and the script gets interrupted.

The proposed change worked for me too.

**Comment by dcooper16** [comment]
This seems to be the same issue as #1754. While most versions of bash return 0 for `$((16#))`, there is at least one (possibly new) version that returns an error. So, it seems that we now need to avoid using `$((16#$x))` if `$x` might be empty.

I found a few places in testssl.sh that use `$((16#...))`:
```
hex2dec() {
     echo $((16#$1))
}

f5_hex2ip() {
     debugme echo "$1"
     echo $((16#${1:0:2})).$((16#${1:2:2})).$((16#${1:4:2})).$((16#${1:6:2}))
}

f5_port_decode() {
     ...
     echo $((16#${tmp:2:2}${tmp:0:2}))  # reverse order and convert it from hex to dec
}
```
I'm not familiar with `f5_hex2ip()` and `f5_port_decode()`, so I don't if any of the 5 uses of `$((16#...))` in these functions could be involve empty strings.

I tried `$((0x))` on a couple of different computers, and it worked on both. However, it might be safer to do something like:
```
if [[ -n "$1" ]]; then
     echo $((16#$1)) # or echo $((0x$1))
else
     echo 0
fi
```

**Comment by drwetter** [comment]
Thanks @dcooper16 : while your sounds reasonable: @Enes13 + @todmephis  : may I have the bash version (``bash --version``) please and OS?

**Comment by eny13** [comment]
Bash Version:
```
$ /usr/bin/env bash --version
GNU bash, version 5.1.0(1)-rc2 (x86_64-pc-linux-gnu)
```
OS: Kali GNU/Linux Rolling

**Comment by todmephis** [comment]
> Hi @todmephis ,
> 
> this is not a general error. And unfortunately there's little information so that I can reproduce that.
> 
> Would you be so kind to provide the details which were asked in the template?
> 
> Thx, Dirk

Sorry, my bad.

```
$ uname -a
Linux Kalynx 5.9.0-kali2-amd64 #1 SMP Debian 5.9.6-1kali1 (2020-11-11) x86_64 GNU/Linux
```

```
 $ testssl.sh -b 2>/dev/null | head -4 | tail -2
    testssl.sh       3.0rc4 from https://testssl.sh/dev/
    (b4bc334 2019-02-23 12:49:45 -- )
```

```
$ testssl.sh -b 2>/dev/null | awk -F':' '/openssl/ { print $2}'
$PWD/bin/openssl.Linux.x86_64
```

```
$bash --version
GNU bash, version 5.1.0(1)-rc2 (x86_64-pc-linux-gnu)
```

To scan my host I just did:

```
$ testssl.sh --csv <IP:PORT>
```
Thanks a lot, let me know if you need any further detail :)

**Comment by drwetter** [comment]
@todmephis: the rc4 from 3.0 is rather old. I couldn't reproduce that with 3.0.4. Don't know whether kali rolling still uses the old version. Please do a git pull or a fresh git clone and and it should be fine as it was fixed in #1759. If not please speak up again ;-)

**Comment by drwetter** [comment]
I cannot reproduce anything on debian testing using bash _5.1.0(1)-rc1_.

@Enes13 : whatever version of testssl.sh you're using: for 3.0 see comment above. For 3.1dev: It is was fixed in #1758.
Thus I am closing this. If you have new info, I am happy to reopen the issue.

For everyone else who's reading: PLEASE USE THE TEMPLATE. 

@dcooper16 : You're were right with your analysis that this happened when the passed variable was empty. If any calls of  ``hex2dec()`` will give a similar error in the future I'd first probably would do the check before calling ``hex2dec()``.
